src/main.cpp

```diff
@@ -17,37 +34,66 @@ int main(int argc, char** argv) {
     std::string cmd = "";
     for (auto i = 0; i < argc; ++i)
         cmd += std::string(i == 0 ? "" : " ") + argv[i];
+
     setenv("HYPRLAND_CMD", cmd.c_str(), 1);
     setenv("XDG_BACKEND", "wayland", 1);
     setenv("_JAVA_AWT_WM_NONREPARENTING", "1", 1);
     setenv("MOZ_ENABLE_WAYLAND", "1", 1);
     setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
 
+
     // parse some args
     std::string configPath;
-    bool        ignoreSudo = false;
-    for (int i = 1; i < argc; ++i) {
-        if (!strcmp(argv[i], "--i-am-really-stupid"))
+    bool ignoreSudo = false;
+
+    // Opinion: All of this code is a mess. Try to do something with that, folks? Pretty please?
```

Comment:
all of what? argument parsing? it is, cuz nobody really cared a lot as we do very few args

Reply:
Good thing to not care about the code, even which is possibly insecure, right?

Reply:
You're trying to find a hole in argument parsing, which assumes the attacker already knows you're running Hyprland.

In that case, running `hyprctl dispatch exec "rm -rf ~"` is already miles easier.

Besides, IDEK what your problem here is, as hyprland does not execute any of the args. The CVE you reference does not make sense here.